### PR TITLE
Network dialog

### PR DESCRIPTION
**Pull request by Bhupesh-mfsi**
#1140 Is also included in the this PR. I will delete the PR #1140 after this merge

### CODE REVIEW

File: packages/messenger-widget/src/components/ConfigureProfile/context/ConfigureProfileContext.tsx [hunk 1]
```diff
@@ -50,6 +52,21 @@ export const ConfigureProfileContextProvider = (props: { children?: any }) => {
         namingServices[0].name,
     );
 
+    const { displayName } = useContext(AuthContext);
+    const { dm3Configuration } = useContext(DM3ConfigurationContext);
+
+    useEffect(() => {
+        setExistingEnsName(
+            displayName
+                ? !displayName.endsWith(dm3Configuration.addressEnsSubdomain) &&
```
Comment (AlexNi245) [inline packages/messenger-widget/src/components/ConfigureProfile/context/ConfigureProfileContext.tsx:61]: please simplify this statement

File: packages/messenger-widget/src/components/Preferences/NormalView.tsx [hunk 2]
```diff
@@ -11,20 +11,33 @@ export function NormalView() {
...
+    // const [optionChoosen, setOptionChoosen] = useState<any>(null);
 
     /**
      *  Opens DM3 profile configuration by default if user clicked
      *  on "Configure Profile" button
      */
     useEffect(() => {
         if (showProfileConfigurationModal) {
-            setOptionChoosen(preferencesItems[1]);
+            setPreferencesOptionSelected(preferencesItems[1]);
```
Comment (AlexNi245) [inline packages/messenger-widget/src/components/Preferences/NormalView.tsx:27]: what is prefernce item[1]?

File: packages/messenger-widget/src/utils/enum-type-utils.ts [hunk 3]
```diff
@@ -33,3 +33,15 @@ export enum SiweValidityStatus {
     VALIDATED = 'VALIDATED',
     ERROR = 'ERROR',
 }
+
+export enum ProfileType {
+    DM3_NAME = 'dm3Name',
+    OWN_NAME = 'ownName',
```
Comment (AlexNi245) [inline packages/messenger-widget/src/utils/enum-type-utils.ts:39]: what is ownName?

File: packages/messenger-widget/src/hooks/userProfile/useDm3UserProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,688 @@
...
+                        ensNameProfile.profile) ||
+                (!isLocalStorageData && ensNameProfile)
+            ) {
+                isEnsNameActive = true;
+            }
+
+            // 1. local storage has ds node and login account profile matches with localsotrage profile
+            // 2. OR local storage is empty & DS from logged in profile has to be checked for updated profile
+            if (
+                (isLocalStorageData &&
```
Comment (AlexNi245) [inline packages/messenger-widget/src/hooks/userProfile/useDm3UserProfile.tsx:218]: this sequence can be extracted to a function

File: packages/messenger-widget/src/hooks/userProfile/useDm3UserProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,688 @@
...
+
+            // filter out duplicate nodes
+            const profileDsNodes = dsNodeFromLocalStorage
+                ? dsNodeFromLocalStorage
+                : dsNodesFromProfile;
+            // const profileDsNodes = dsNodesFromProfile.filter((d) => !storageNodes.find((s) => s.dsName));
+
+            const updatedDsNodes = await fetchUserProfileForAllNames(
+                profileDsNodes,
+                dsNodeFromLocalStorage ? true : false,
```
Comment (AlexNi245) [inline packages/messenger-widget/src/hooks/userProfile/useDm3UserProfile.tsx:299]: !!dsNodeFromLocalStorage

File: packages/messenger-widget/src/components/Preferences/NormalView.tsx (hunk 2 above)
Comment (Bhupesh-mfsi) [reply packages/messenger-widget/src/components/Preferences/NormalView.tsx:27]: preferences items is an array which contains all properties to be shown on UI like :- 
1. Configure profile
2. Network
3. Notification etc....

Here item[1] means by default "Configure Profile" is selected

File: packages/messenger-widget/src/components/ConfigureProfile/context/ConfigureProfileContext.tsx (hunk 1 above)
Comment (Bhupesh-mfsi) [reply packages/messenger-widget/src/components/ConfigureProfile/context/ConfigureProfileContext.tsx:61]: ok will do it

File: packages/messenger-widget/src/components/ConfigureProfile/MobileView.tsx [hunk 6]
```diff
@@ -50,59 +77,245 @@ export function MobileView() {
...
+                        {/* Delete icon */}
+                        <img
+                            className="ms-4 pointer-cursor del-icon"
+                            src={deleteIcon}
+                            alt="remove"
+                            onClick={() => setShowDeleteConfirmation(true)}
+                        />
+                    </p>
+
+                    {!isProfileUpdatedForDm3Name() && (
```
Comment (AlexNi245) [inline packages/messenger-widget/src/components/ConfigureProfile/MobileView.tsx:186]: use the useMemo hook to store the results of functions like isProfileUpdatedForDm3Name

File: packages/messenger-widget/src/components/ConfigureProfile/context/ConfigureProfileContext.tsx (hunk 1 above)
Comment (AlexNi245) [reply packages/messenger-widget/src/components/ConfigureProfile/context/ConfigureProfileContext.tsx:61]: please simplify this statement

**Pull request merged**